### PR TITLE
dependency_management.md: document go.opentelemetry.io/otel version update is blocked

### PR DESCRIPTION
**Pull request by chaochn47**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

XREF: https://github.com/etcd-io/etcd/pull/15810

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @chaochn47

File: Documentation/contributor-guide/dependency_management.md [hunk 1]
```diff
@@ -81,6 +81,8 @@ There is a compatible issue between etcd and gRPC 1.52.0, and there is a pending
 The plan is to remove the dependency on some grpc-go's experimental API firstly, afterwards try to bump it again. Please get more details in
 [issues/15145](https://github.com/etcd-io/etcd/issues/15145).
 
+`go.opentelemetry.io/otel` version update is indirectly blocked due to this gRPC issue. Please get more details in [pull/15810](https://github.com/etcd-io/etcd/pull/15810)
```
Comment (ahrtr) [inline Documentation/contributor-guide/dependency_management.md:84]:
```suggestion
`go.opentelemetry.io/otel` version update is indirectly blocked due to this gRPC issue. Please get more details in [pull/15810](https://github.com/etcd-io/etcd/pull/15810).
```

**Comment by ahrtr** [comment]
Just a minor comment.

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks for taking care of maintenance this week @chaochn47 👍🏻

**Review by ahrtr — APPROVED** [review]

**Pull request merged**